src/utils/drbUtils.ts

```diff
@@ -0,0 +1,151 @@
+import type { SearchFilters, SearchParams } from "../types/searchTypes"
+import type { DRBQueryParams, DRBFilters } from "../types/drbTypes"
+import { DRB_RESULTS_PER_PAGE } from "../config/constants"
+
+const mapSearchFieldToDRBField = {
+  all: "keyword",
+  contributor: "author",
+  standard_number: "standardNumber",
+  title: "title",
+  subject: "subject",
+  date: "date",
+}
+
+/**
+ *  Given a keyword and a search field, format and return a keyword query string expected by the DRB API
+ */
+function getDRBKeywordQuery(keywords = "*", field = "keyword"): string {
+  return `${field}:${keywords}`
+}
+
+/**
+ *  Given a hash of SearchQueryParams, format and return an advanced field query string expected by the DRB API
+ */
+function getDRBAdvancedQuery(params: SearchParams): string {
+  return ["contributor", "title", "subject"]
+    .map((fieldType) => {
+      const fieldValue = params[fieldType]
+      return (
+        fieldValue && `${mapSearchFieldToDRBField[fieldType]}:${fieldValue}`
+      )
+    })
+    .filter((str) => str)
+    .join(",")
+}
+
+/**
+ *  Given a hash of SearchFilters, returns an array of DRBFilters as expected by the DRB API
+ */
+function mapSearchFiltersToDRBFilters(filters: SearchFilters = {}): DRBFilters {
+  let drbFilters = [] as DRBFilters
```

Comment:
Did you run into issues with the following syntax? `let drbFilters: DRBFilters = []`

Reply:
No issues with that, I believe it's just an alternative syntax. Replaced with the one you proposed since it looks a bit cleaner.